python_ta/checkers/inconsistent_or_missing_returns_checker.py

```diff
@@ -0,0 +1,82 @@
+"""
+Check for inconsistent return statements in functions and missing return statements in non-None functions.
+"""
+
+from typing import Optional
+
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.lint import PyLinter
+
+from python_ta.cfg import ControlFlowGraph
+
+
+class InconsistentReturnChecker(BaseChecker):
+
+    name = "inconsistent-or-missing-returns"
+    msgs = {
+        # as R1710 is being disabled, we replace it with an identical message
+        "R9710": (
+            """This function has at least one case where the function body will execute without ending in an explicit return statement.
+            You should check your code to make sure every possible execution path through the function body ends in a return statement.
+            Note: one common source of this error is if you're using if statements without an explicit else branch.
+            In this case, you should consider revising your code to either add an else branch,
+            or, if you are confident that the if and elif conditions cover all possible cases,
+            "you can convert the final "elif " into an " else ".",""",
+            "inconsistent-returns",
+            "Used to replace R1710 message inconsistent-return-statements",
+        ),
+        "R9711": (
+            "Missing return statement in function",
+            "missing-return-statements",
+            "Used when a function does not have a return statement and whose return type is not None",
+        ),
+    }
+
+    def __init__(self, linter: Optional[PyLinter] = None) -> None:
+        super().__init__(linter=linter)
+
+    def visit_functiondef(self, node) -> None:
+        """Visit a function definition"""
+        self._check_return_statements(node)
+
+    def _check_return_statements(self, node) -> None:
+        """
+        Construct a CFG from the function. Check for inconsistent returns if there are
+        multiple return statements, and missing return statements if there are none.
+        """
+        if (
+            node.returns is None
+            or isinstance(node.returns, nodes.Const)
+            and node.returns.value is None
+        ):
+            return
+
+        # get the end of CFG
+        cfg = ControlFlowGraph()
```

Comment:
You shouldn't need to create a new CFG; there is a PythoNTA node visitor that creates a CFG before any of the checkers run. Try looking at some of the existing custom checkers that are CFG-based to see how they access nodes.

Reply:
Hi Professor, I checked that a function definition node only contains a reference to the start of the function (cfg_block attribute), so I do need to invoke get_blocks_postoreder() method in ControlFlowGraph to access other blocks of the CFG. Some other CFG-based checkers, such as redundant_assignment_checker, use the same approach.

However, if we make a slight change in CFGVisitor and append each module definition and function definition with a reference to its corresponding CFG graph, we can directly access the whole CFG inside the checker. I was wondering if you'd like to consider this change, and then we can probably also refactor the existing checkers as well.

Reply:
Great research, @Raine-Yang-UofT. Please modify the `CFGVisitor` class as you suggested so that the `ControlFlowGraph` can be accessed. However, only use the new attribute in this checker; we'll leave the work of refactoring the other checkers as a separate PR.